qdax/core/containers/spea2_repertoire.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import flax
+import jax
+import jax.numpy as jnp
+
+from qdax.core.containers.ga_repertoire import GARepertoire
+from qdax.types import Fitness, Genotype
+
+
+class SPEA2Repertoire(GARepertoire):
+    """Repertoire used for the SPEA2 algorithm.
+
+    Inherits from the GARepertoire. The data stored are the genotypes
+    and there fitness. Several functions are inherited from GARepertoire,
+    including size, save, sample.
+    """
+
+    num_neighbours: int = flax.struct.field(pytree_node=False)
```

Comment:
It does not initialise it as a PyTree, it tells `flax.struct.PyTreeNode` that `num_neighbours` does not have to be considered as a PyTree in this class.

Indeed, `SPEA2Repertoire` inherits from `flax.struct.PyTreeNode` which is relevant because most of its data are PyTree, but we do not want to loose this just because of the `num_neigbours` that plays like a static argument here. Hopefully, the feature `flax.struct.field` allows us to do this!